packages/template-retail-react-app/app/hooks/use-auth-modal.js

```diff
@@ -82,55 +82,57 @@ export const AuthModal = ({
         }
 
         return {
-            login: (data) =>
-                login.mutateAsync(
-                    {username: data.email, password: data.password},
-                    {
-                        onSuccess: onLoginSuccess,
-                        onError: (error) => {
-                            const message = /Unauthorized/i.test(error.message)
-                                ? formatMessage(LOGIN_ERROR)
-                                : formatMessage(API_ERROR_MESSAGE)
-                            form.setError('global', {type: 'manual', message})
-                        }
+            login: async (data) => {
+                try {
+                    const res = await login.mutateAsync({
+                        username: data.email,
+                        password: data.password
+                    })
+                    if (res) {
+                        onLoginSuccess()
                     }
```

Comment:
I think `res` is a promise and it is always defined.

```suggestion
                    onLoginSuccess()
```